### PR TITLE
feat: add package name to PURL conversion

### PR DESCRIPTION
**Pull request by Churro**
### Changes

Adds a conversion util that translates package and ecosystem names to PURL identifiers that comply with the PURL specification (see [here ](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst) and [here](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst)) and the [osv.dev reference implementation](https://github.com/google/osv.dev/blob/edacbd3c32b1d632bad8c5b506a14d0850b2e537/osv/purl_helpers.py#L18-L30).

Currently, db lookups only work in limited cases, e.g. with `npm` (where the purl pendant is also `npm`) and package names that are already purl-compliant without translation. As of now, npm package names with scopes won't yield results, as well as any Maven package name incl `:`. E.g., `commons-fileupload:commons-fileupload` will be queried with purl identifier `pkg:maven/commons-fileupload:commons-fileupload`, whereas `pkg:maven/commons-fileupload/commons-fileupload` would be correct.

### CODE REVIEW

File: packages/osv-offline-db/src/lib/purl-helper.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  NuGet: 'nuget',
+  npm: 'npm',
+  Packagist: 'composer',
+  PyPI: 'pypi',
+  RubyGems: 'gem',
+} as const;
+
+function urlEncode(packageName: string): string {
+  const parts = packageName.split('/');
+  return parts.map(encodeURIComponent).join('/');
```
Comment (JamieMagee) [inline packages/osv-offline-db/src/lib/purl-helper.ts:18]: Can you add a test to exercise `encodeURIComponent`?

**Review by JamieMagee — APPROVED** [review]
1 small comment, otherwise ready to go.

**Comment by JamieMagee** [comment]
Forgot to add, the test might need to have the `.unit.spec.ts` ending. I set up this repo to have integration tests that make HTTP calls in files that end in `.int.spec.ts`, and unit tests in `.unit.spec.ts`.

**Comment by JamieMagee** [comment]
1 thing I just remembered: can you also check the license that the Hex and Packagist vulnerability data is provided under. Most require attribution, so I've added the information to all the `README.md` files.

It may also need to be added to any PRs Renovate opens based on this data.

**Comment by Churro** [comment]
For Hex and Packagist the upstream source is the [GitHub Advisory Database](https://github.com/github/advisory-database) ([CC-BY 4.0](https://github.com/github/advisory-database/blob/main/LICENSE.md)), which is already listed in `README.md`. Thank you for pointing this out. I agree that we should be mindful about this and I'm going to bake a license reference into renovate PR messages as well.

File: packages/osv-offline-db/src/lib/purl-helper.ts (hunk 1 above)
Comment (Churro) [reply packages/osv-offline-db/src/lib/purl-helper.ts:18]: Already there:
https://github.com/renovatebot/osv-offline/blob/49b7591e6be87fcb7648554e3d343f20398b83b8/packages/osv-offline-db/src/lib/purl-helper.unit.spec.ts#L12

I also went over all databases but it seems npm is the only one needing this.

**Pull request merged**